apps/mysql.go

```diff
@@ -15,13 +15,75 @@
 package apps
 
 import (
-	"fmt"
-	"strings"
+	"os"
 
 	"github.com/GoogleCloudPlatform/ops-agent/confgenerator"
 	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/fluentbit"
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/otel"
+
+	"fmt"
+	"strings"
 )
 
+type MetricsReceiverMySQL struct {
+	confgenerator.ConfigComponent `yaml:",inline"`
+
+	confgenerator.MetricsReceiverShared `yaml:",inline"`
+
+	Endpoint string `yaml:"endpoint" validate:"omitempty,hostname_port|file"`
+
+	Password  string `yaml:"password" validate:"omitempty"`
+	Username  string `yaml:"username" validate:"omitempty"`
+	Transport string `yaml:"transport" validate:"omitempty"`
+}
+
+const defaultMySQLTCPEndpoint = "localhost:3306"
+const defaultMySQLUnixEndpoint = "/var/run/mysqld/mysql.sock"
+
+func (r MetricsReceiverMySQL) Type() string {
+	return "mysql"
+}
+
+func (r MetricsReceiverMySQL) Pipelines() []otel.Pipeline {
+	if r.Endpoint == "" {
+		if _, err := os.Stat(defaultMySQLUnixEndpoint); err != nil {
```

Comment:
This means that the config generation is non-hermetic and can't be unit tested. Do we know that there are environments where we can't default to the unix socket? (What are they? Windows?) Even if those environments do require TCP, they would also presumably require a password, so I think we can just unconditionally default to TCP.

Also see above; this can be racy with MySQL startup.

Reply:
bit confused on the desired behavior you're looking for. Are you saying we should just default to Unix sockets without trying to os.Stat it and just not bother with TCP defaults?

Reply:
I'm asking if that works in all the situations where the current code does. I think it does but I'm not sure about any edge cases.

Reply:
From what I'm seeing, yes I think that'll work. There seems there was a way to set no password for mysql prior to 5.7 for windows machines but that it was removed.

Reply:
Looks like this has been resolved.